### DISCUSSION TITLE
Is there anyway to make "always on top" no the default for Gnome/mutter

### DISCUSSION THREAD

**Discussion by escape0707** [xiaoyifang/goldendict-ng#1277] [Q&A] [answered]
I personally use <kbd>Ctrl</kbd>+<kbd>D</kbd> for goldendict shortcut (`sh -c "goldendict $(wl-paste)"`), mostly when viewing lecture videos or webpages for looking up unfamiliar words. But after the search, I need to return to the video player or the browser. I prefer to use <kbd>Alt</kbd>+<kbd>Tab</kbd> to do this, but the windows of goldendict is always marked with `Always on top` on for mutter. Can I somehow disable this behavior?

I can use <kbd>Esc</kbd> to close the main window to work around that, but then I can't easily <kbd>Alt</kbd>+<kbd>Tab</kbd> again and begin searching for another new word.

**Comment by escape0707** [comment]
Never mind, I used `env GOLDENDICT_FORCE_WAYLAND=1 goldendict` to force it to run under Wayland. And it doesn't make itself `Always on top` for now at least.

**Comment by xiaoyifang** [comment]
maybe this will help
https://github.com/xiaoyifang/goldendict-ng/discussions/1072

**Reply by escape0707** [reply]
Thanks for replying. I've just restored to use XWayland and gave the `-m` and `-s` option a try, but it seems that they can't solve my problem. With either option, the first time I revoke my keyboard shortcut for GoldenDict, the first query is shown in the main window, only with `-s` will GoldenDict show queries starting from the second one in the scan popup window. The main window is still marked as `Always on top` no matter which option is used.

And because I'm using Gnome / Mutter with Wayland, scan popup can't be easily dismissed, which is also the reason why I stick to `alt+tab`ing between GoldenDict and the application windows in the first place. So I prefer looking up words in the main windows over the scan pop up windows. The good thing is, at least running GoldenDict-ng under Wayland solves all my problems (for now).

**Comment by escape0707** [comment]
@xiaoyifang Sorry, but please allow me to reopen this discussion or potentially transform this into an issue, as I recently experienced malfunction of displaying under Wayland. 

I haven't tried to figure out a way to stably reproduce it but, sometimes I closed a previous run of GoldenDict-ng under Wayland, then rerun it, it will not display anything, sometimes it will even get displayed as a transparent rectangle or pure black rectangle. And client side decoration won't be displayed, either.

I'm forced to use Xwayland under those circumstances, but this bring back the issue that the program starts with `Always on top` ticked.

I have no idea if this has anything to do with me changing to some third party gtk/icon themes (Arc + Papirus).

**Reply by escape0707** [reply]
This seems to be a mutter issue that is fixed recently. I'm not facing this issue anymore.

**Comment by escape0707** [comment, marked as answer]
As to the original issue that when using `goldendict <word to look up>`, the fired up goldendict window will always be launched with the flag "Always on Top", I figured out that it's actually a config [here](https://github.com/xiaoyifang/goldendict-ng/blob/2cd6ed07561cbd6c241856a91ee86e48d726f7e8/src/ui/mainwindow.ui#L491) that is on by default. I didn't notice this because I always have the top menu hided.

So go to the top menu - View - uncheck Always on Top will solve my original issue.

Closing this.
